### ISSUE TITLE
Share rooms with "full" admin rights / have rooms owned by groups

### ISSUE THREAD

**Issue by sistason** [sualko/cloud_bbb#250]
Hi,

the BBB plugin for Nextcloud is amazing and does most of the things I want. Thanks for you work! :)

I have the current **problem**, for which my naive solution does not work. I'd like to know if this would be possible some other way.

For an organisation, I use user_saml for SSO login and I sync accounts via a sync-script, which creates users, group_folders, and should also create BBB rooms for all "groups" in the SSO.
Seeing the BBB plugin work and seeing the DB table structure, I presumed I can create all group-rooms as rooms of the uid=admin and share the room "with full admin rights" to all members of the respective group.

Sadly, they cannot edit the room settings, which is only possible by the _single_ uid the room belongs to. As I want to dynamically generate rooms based on SSO groups, and those group "manage" the room and the members of that group are equal and dynamic, it does not work for me that there can only be one ~~highlander~~ room admin actually able to admin the room.

**Question**:
Is there a technical (BBB-) reason why users being shared a room with Administrator rights cannot edit the room settings?
Do you see some other way to generate rooms to be used (e.g. administered) by multiple users / groups?

**Solutions?**:
Ideally, I'd like to either that rooms shared with Administrator rights can be edited by those users, or that there is a checkbox "Admins can edit settings of this room" or that there is a "share as Settings-Administrator" so users are able to change settings.

I think in the roomShareController->isUserAllowed there is the check to room.uid == user.uid, so maybe there could be a more nuanced permission model?

 Thank you for any help or pointers in a direction I can implement this functionality.

**Comment by sualko** [comment]
> Is there a technical (BBB-) reason why users being shared a room with Administrator rights cannot edit the room settings?

It should be possible for an admin to edit a room. So what you mean by "can't edit the room settings"?

**Comment by sistason** [comment]
Oh wow, it's all my fault for misreading the problem at hand, sorry there is no issue... -.-

I could not change any of the options of a room, even though I was an admin.
![image](https://github.com/sualko/cloud_bbb/assets/2707108/ef1a0be5-736d-4657-8c55-e719365b1261)

Because of the `You are not allowed to change this option, because this room is shared with you.`, I expected this to be intended for all of the option(s) and therefore opened this ticket.

Turns out the PUT request for the room threw a 500 without an error message in the GUI, since I generated all rooms in the database without a default welcome message, so `null`, which then triggered `Argument #3 ($welcome) must be of type string, null given`. Setting the welcome message removes the 500 and I can edit all options of a room, when it is shared to me as admin... :)

So we could just close this issue here, or (if you agree/have time for details):
- Rephrase `You are not allowed to change this option` to `You are not allowed to share this room further`, or draw a border around Participants and the Notice, to not think that the information might be valid for all following options. But this might be just a Me-problem :)
- Alter the table `oc_bbb_rooms` to enforce a default for the `welcome` value, or cast Null to an empty string in the frontend instead of throwing an exception.
